java/src/org/openqa/selenium/support/decorators/WebDriverDecorator.java

```diff
@@ -293,7 +309,8 @@ protected final <Z> Z createProxy(final Decorated<Z> decorated, Class<Z> clazz)
             || decoratedInterfaces.contains(method.getDeclaringClass())) {
           return method.invoke(decorated, args);
         }
-        if (originalInterfaces.contains(method.getDeclaringClass())) {
+        if (originalInterfaces.contains(method.getDeclaringClass()) || findInterfaceByMethod(originalInterfaces,
```

Comment:
Can you help me understand the need for `findInterfaceByMethod`? I might be missing a use case this helps cover.

Reply:
revised and removed